src/handlers/Registry.ts

```diff
@@ -1,19 +1,19 @@
-import { type Context, type Event } from "ponder:registry";
+import { type Context } from "ponder:registry";
 import { resolvers } from "ponder:schema";
 import { domains } from "ponder:schema";
 import { type Hex, zeroAddress } from "viem";
 import { ROOT_NODE, encodeLabelhash, makeSubnodeNamehash } from "../lib/ens-helpers";
 import { makeResolverId } from "../lib/ids";
-import { NsReturnType } from "../lib/plugins";
 import { upsertAccount } from "../lib/upserts";
 
-type NsType<T extends string> = NsReturnType<T, "/eth">;
-
-export async function setup({ context }: { context: Context }) {
+/**
+ * Initialize the ENS root node with the zeroAddress as the owner.
+ */
+export async function handleRootNodeCreation({ context }: { context: Context }) {
```

Comment:
agreed with lightwalker's naming concern, and apologies for the nitpick, but how about calling this `setupRootNode`